libkbfs/folder_branch_ops.go

```diff
@@ -886,6 +1119,26 @@ func (fbo *folderBranchOps) commitFlushedMD(
 
 		fbo.log.CDebugf(ctx, "Prefetch for revision %d complete; commiting",
 			rev)
+	case keybase1.FolderSyncMode_PARTIAL:
+		// For partially-synced TLFs, wait for the partial sync to
+		// complete, or for an update to happen.
+		lState := makeFBOLockState()
+		fbo.kickOffPartialSyncIfNeeded(ctx, lState, rmd)
```

Comment:
So if this is about to happen while a partial sync is already happening, then `fbo.latestMergedUpdated` would be closed and the previous sync thus gets automatically canceled. Is my understanding correct?

Reply:
Correct.  As soon as `setLatestMergedRevisionLocked` is called (which happens before this method is called in both cases), the previous sync gets canceled because of the channel close.